ksql-engine/src/main/java/io/confluent/ksql/function/udtf/array/ExplodeFunctionFactory.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udtf.array;
+
+import io.confluent.ksql.function.KsqlTableFunction;
+import io.confluent.ksql.function.TableFunctionFactory;
+import io.confluent.ksql.function.udf.UdfMetadata;
+import io.confluent.ksql.name.FunctionName;
+import io.confluent.ksql.util.KsqlException;
+import java.util.List;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Schema.Type;
+
+public class ExplodeFunctionFactory extends TableFunctionFactory {
```

Comment:
I'm not sure that would be a good idea. What's the advantage of using annotation framework for built in functions? I don't think we should be deliberately taking a performance hit where we don't have to.

Reply:
I think dogfooding is both an excellent source of documentation and a forcing function to improve our framework. If the the annotation performance overhead is an issue for non-builtin UDFs, then we should fix it for all UDFs (not just ours).

Reply:
"non built in UDFs", i.e. "user defined, user defined functions?". I think this illustrates why our naming around this is a bit nuts. Why not just call them "built in functions" and "user defined functions". Seems a lot clearer ;)

Reply:
I'm with @agavra on this for three reasons:
1. Why have two ways of doing a thing when one is enough?
1. Dogfooding is the main way we find the holes in the UDF framework: we look to implement UDF/UDAF/UDTF X and work out why can't because of Y and Z, so we extend the framework as needed and everyone benefits.
1. We don't know the performance hit of using reflection is statistically relevant. It mostly likely isn't and won't be for some time.  If/when we get to the point that it is, then we should look to see if we can solve this for everyone, as @agavra said.

Reply:
We will have to agree to disagree on that one.